src/components/ReportActionItem/ReportPreview.tsx

```diff
@@ -162,6 +170,28 @@ function ReportPreview({
         [chatReport?.isOwnPolicyExpenseChat, policy?.harvesting?.enabled],
     );
 
+    const confirmPayment = (type?: PaymentMethodType | undefined) => {
+        if (!type) {
+            return;
+        }
+        setPaymentType(type);
+        setRequestType('pay');
+        if (ReportUtils.hasHeldExpenses(iouReport?.reportID)) {
+            setIsHoldMenuVisible(true);
+        } else if (chatReport && iouReport) {
+            IOU.payMoneyRequest(type, chatReport, iouReport, false);
+        }
+    };
+
+    const confirmApproval = () => {
```

Comment:
Should we create a utility function to avoid code repetition?

Reply:
In this these function, we also have logic to set state of the component so we shouldn't create a util for this.

Reply:
> we also have logic to set state of the component so we shouldn't create a util for this.

Do you mean we shouldn't pass setIsHoldMenuVisible,... to the util function? Could you explain more why we shouldn't do that

Reply:
I mean we shouldn't create a util function for this case since this function also controls the state of a component.

Reply:
And Since we only use this logic for report preview and `MoneyReportHeader`, the duplicate is acceptable.